packages/web-sdk/src/transports/otlp/transport.ts

```diff
@@ -0,0 +1,120 @@
+import { BaseTransport, createPromiseBuffer, isArray, PromiseBuffer, TransportItem, VERSION } from '@grafana/faro-core';
+
+import { OtelPayload, OtelTransportPayload } from './payload';
+import type { OtlpHttpTransportOptions } from './types';
+
+const DEFAULT_BUFFER_SIZE = 30;
+const DEFAULT_CONCURRENCY = 5; // chrome supports 10 total, firefox 17
+const DEFAULT_RATE_LIMIT_BACKOFF_MS = 5000;
+
+export class OtlpHttpTransport extends BaseTransport {
+  readonly name = '@grafana/faro-web-sdk:transport-otlp-http';
+  readonly version = VERSION;
+
+  private readonly promiseBuffer: PromiseBuffer<Response | void>;
+  private readonly rateLimitBackoffMs: number;
+
+  private disabledUntil: Date = new Date();
+
+  constructor(private options: OtlpHttpTransportOptions) {
+    super();
+    this.rateLimitBackoffMs = options.defaultRateLimitBackoffMs ?? DEFAULT_RATE_LIMIT_BACKOFF_MS;
+
+    this.promiseBuffer = createPromiseBuffer({
+      size: options?.bufferSize ?? DEFAULT_BUFFER_SIZE,
+      concurrency: options?.concurrency ?? DEFAULT_CONCURRENCY,
+    });
+  }
+
+  override getIgnoreUrls(): Array<string | RegExp> {
+    const { tracesURL = '', logsURL = '', metricsURL = '' } = this.options;
+    return [tracesURL, logsURL, metricsURL].filter(Boolean);
+  }
+
+  send(item: TransportItem | TransportItem[]): void {
+    const otelPayload = new OtelPayload(undefined, this.internalLogger);
+    const items = isArray(item) ? item : [item];
+
+    items.forEach((item) => otelPayload.addResourceItem(item));
+    this.sendPayload(otelPayload.getPayload());
+  }
+
+  private sendPayload(payload: OtelTransportPayload): void {
+    try {
+      if (this.disabledUntil > new Date(Date.now())) {
```

Comment:
does current backoff make sense now that there are 3 receiving URLs per signal? maybe it should be separate for each? :thinking:  could be separate backend entirely, eg sending logs to Loki and traces to Tempo

Reply:
You're right it would be better to backoff per endpoint